src/application_config/configuring_base.py

```diff
@@ -21,49 +22,45 @@
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase(ABC):
-    """Base class for all ConfigSection classes"""
+class ConfigSectionBase:
+    """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
 
 
-# The next line has a type: ignore because MyPy has a problem with @abstractmethod
-# If you comment the line with @abstractmethod, you can do type checking
 @dataclass(frozen=True)
-class ConfigBase(ABC):
+class ConfigBase:
     """Base class for main Config class"""
 
-    @staticmethod
-    @abstractmethod
-    def get_app_basename() -> str:
-        """Return the application base name.
-
-        This name, with a preceding dot, will also be the folder name in the home directory
-        that will store the config.
-        """
+    @classmethod
+    def default_config_foldername(cls: type[TConfig]) -> str:
+        """Return the class name without 'Config' and with a preceding dot as default for the folder that will hold the config file."""
+        return f".{cls.__name__.replace('Config', '')}"
```

Comment:
Personally (mostly due to linux) I would prefer a default with lowercase, maybe split on the capital letters?

so `SomeExampleConfig` -> `some_example`

Would be something like:
```python
re.sub(r"([A-Z])", r"_\1", cls.__name__.replace('Config', ''))
```

Reply:
Done.